plothist/plotters.py

```diff
@@ -469,107 +321,237 @@ def plot_error_hist(hist, ax, **kwargs):
+        The label for the x-axis. Default is "x1".
+    x1_label : str, optional
+        The label for the first histogram. Default is "x1".
+    x2_label : str, optional
+        The label for the second histogram. Default is "x2".
+    comparison : str, optional
+        The type of comparison to plot ("ratio" or "pull"). Default is "ratio".
+    comparison_ylim : tuple or None, optional
+        The y-axis limits for the comparison plot. Default is None.
+    ratio_uncertainty : str, optional
+        How to treat the uncertainties of the histograms when comparison = "ratio" ("uncorrelated" for simple comparison, "split" for scaling and split hist_1 and hist_2 uncertainties)
+
+    Returns
+    -------
+    ax : matplotlib.axes.Axes
+        The axes with the plotted comparison.
+
+    See Also
+    --------
+    compare_two_hist : Compare two histograms and plot the comparison.
+
+    """
+    if not np.all(hist_1.axes[0].edges == hist_2.axes[0].edges):
+        raise ValueError("The bins of the compared histograms must be equal.")
 
     np.seterr(divide="ignore", invalid="ignore")
-    ratio = np.where(hist_1.values() != 0, hist_2.values() / hist_1.values(), np.nan)
-    ratio_variance = np.where(
-        hist_1.values() != 0,
-        hist_2.variances() / hist_1.values() ** 2
-        + hist_1.variances() * hist_2.values() ** 2 / hist_1.values() ** 4,
-        np.nan,
-    )
+    if comparison == "ratio":
+        ratio_values = np.where(
+            hist_2.values() != 0, hist_1.values() / hist_2.values(), np.nan
+        )
+        if ratio_uncertainty == "split" :
+            ratio_variance = (np.sqrt(hist_1.variances()) / hist_1.values()) ** 2
+        elif ratio_uncertainty == "uncorrelated":
+            ratio_variance = np.where(
+                hist_2.values() != 0,
+                hist_1.variances() / hist_2.values() ** 2
+                + hist_2.variances() * hist_1.values() ** 2 / hist_2.values() ** 4,
+                np.nan,
+            )
+        else:
+            raise ValueError("ratio_uncertainty not in ['uncorrelated', 'split'].")
+        if ratio_uncertainty == "split" :
+            h2_uncertainty = np.sqrt(hist_2.variances()) / hist_2.values()
+
+    elif comparison == "pull":
+        ratio_values = np.where(
+            hist_2.values() != 0,
+            (hist_1.values() - hist_2.values())
+            / np.sqrt(hist_1.variances() + hist_2.variances()),
+            np.nan,
+        )
+        ratio_variance = np.where(
+            hist_1.values() != 0,
+            1,
+            np.nan,
+        )
+    else:
+        raise ValueError(
+            f"{comparison} not available as a comparison (use ratio or pull)."
+        )
+
     np.seterr(divide="warn", invalid="warn")
 
-    ax_ratio.errorbar(
-        x=hist_1.axes[0].centers,
+    ax.errorbar(
+        x=hist_2.axes[0].centers,
         xerr=None,
-        y=np.nan_to_num(ratio, nan=0),
+        y=np.nan_to_num(ratio_values, nan=0),
```

Comment:
Rename to `comparison_values`. Less confusing, because this is also used in the `"pulls"` instead of `"ratio"` argument.